### ISSUE TITLE
Question abount RL fine-tuning

### ISSUE THREAD

**Issue by zhexilian** [xiaomi-research/recogdrive#2]
Hi authors! @owl-10 

Thanks for your groundbreaking work! I have two questions as follows:

(1) I wonder if you have used any open-source RL framework/toolkit during RL fine-tuning in NAVSIM?

(2) Would you mind to share some technical details on how to train RL in NAVSIM sometime?

Looking forward to your reply!

**Comment by owl-10** [comment]
Thank you for your interest in our work on NAVSIM! I'm happy to provide some insights into your questions.

(1) Regarding your first question, ReCogDrive did not utilize an open-source RL framework. Instead, we used PyTorch Lightning for the training process. For the diffusion fine-tuning part of ReCogDrive, we partially referenced the [DPPO project ](https://github.com/irom-princeton/dppo) and we are grateful for their excellent work.

(2) In response to your second question, we are currently in the process of open-sourcing our code. Due to our company's rigorous code review process, we anticipate the code will be released within the next one to two weeks.

As for the technical details of the RL training in NAVSIM, we use the PDM Score as the reward. The process begins by sampling various trajectories and diffusion chains from different noises, which are initialized from a Gaussian distribution. We then calculate the reward using the NAVSIM simulator. Following this, the advantage is obtained through group computation. The probability density of the trajectories is calculated from the sampled chains, which in turn allows us to compute the policy loss. Finally, a behavior cloning loss is added. More details on the parameters are available in the supplementary materials of our paper.

If you would like to delve deeper into the details, please feel free to send me an email. Perhaps we can connect on WeChat to discuss this further.
